File: Bit_Manipulation/190_reverse_bits/190.cpp
#include <stdint.h>

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t result = 0;
        uint32_t power = 31;
        while (n != 0) {
            result += (n & 1) << power;
            n = n >> 1;
            power -= 1;
        }
        return result;
    }
};

File: Bit_Manipulation/190_reverse_bits/190.rs


impl Solution {
    pub fn reverse_bits(x: u32) -> u32 {
        let mut result: u32 = 0;
        let mut power: u32 = 31;
        let mut x: u32 = x;
        while x != 0 {
            result += (x & 1) << power;
            x = x >> 1;
            power -= 1;
        }
        return result;
    }
}